### PR TITLE
Phase2-hgx307 Update HGCal cell defintion using PR #37505

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Update HGCal cell defintion using PR #37505 extending it for the dd4hep portion

#### PR validation:

Tested using the cfg's in the test area of HGCalCommonData

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by cvuosalo** [comment]
+1

File: Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalCell.cc [hunk 1]
```diff
@@ -80,10 +83,18 @@ static long algorithm(dd4hep::Detector& /* description */, cms::DDParsingContext
   double dy1 = r;
   double dy2 = 0.5 * dy1;
   double dy3 = 1.5 * dy1;
+  double dx5 = (addWaferSeparation == 0) ? 0.0 : waferSeparation * 0.5;
```
Comment (srimanob) [inline Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalCell.cc:86]: Just wonder why don't you arrange dx1....dx7, then dy1 .... dy6.
But this is just to make in order but not necessary.

File: Geometry/HGCalCommonData/test/python/g4OverlapCheckWafer_cfg.py [hunk 2]
```diff
@@ -3,7 +3,7 @@
 
 process = cms.Process("OverlapTest",Phase2C11)
 
-process.load('Geometry.HGCalCommonData.testHGCalWaferTestXML_cfi')
+process.load('Geometry.HGCalCommonData.testHGCalWaferXML_cfi')
```
Comment (srimanob) [inline Geometry/HGCalCommonData/test/python/g4OverlapCheckWafer_cfg.py:6]: After changing, do you still need `testHGCalWaferTestXML_cfi` ?

File: Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalCell.cc (hunk 1 above)
Comment (bsunanda) [reply Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalCell.cc:83]: Done

File: Geometry/HGCalCommonData/test/python/g4OverlapCheckWafer_cfg.py (hunk 2 above)
Comment (bsunanda) [reply Geometry/HGCalCommonData/test/python/g4OverlapCheckWafer_cfg.py:6]: This is testing the earlier cell version. It is good to preserve it till the version V12 is preserved

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by cvuosalo** [comment]
+1

**Comment by srimanob** [comment]
+Upgrade

This PR is to update the cell definition of HGCal. No change is expected with this PR.

**Comment by perrotta** [comment]
+1

**Pull request merged**